### PR TITLE
Better `ComposeScene` naming

### PR DESCRIPTION
**Pull request by MatkovIvan**
- `MultiLayerComposeScene` -> `CanvasLayersComposeScene`
- `SingleLayerComposeScene` -> `PlatformLayersComposeScene`
- Change deprecation level to `ERROR` for old `ComposeScene`

### CODE REVIEW

File: compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/PlatformLayersComposeScene.skiko.kt [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+ * @return The created [ComposeScene].
+ *
+ * @see ComposeScene
+ */
+@InternalComposeUiApi
+fun PlatformLayersComposeScene(
+    density: Density = Density(1f),
+    layoutDirection: LayoutDirection = LayoutDirection.Ltr,
+    size: IntSize? = null,
+    coroutineContext: CoroutineContext = Dispatchers.Unconfined,
```
Comment (JakeWharton) [inline compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/PlatformLayersComposeScene.skiko.kt:70]: Now would be a good time to change this default here and on `CanvasLayersComposeScene`

https://github.com/JetBrains/compose-multiplatform/issues/4799

Comment (MatkovIvan) [reply compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/PlatformLayersComposeScene.skiko.kt:70]: Good point, thanks for reminder

Comment (MatkovIvan) [reply compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/PlatformLayersComposeScene.skiko.kt:70]: After discussions with @igordmn and @m-sasha we didn't agree what should be used as a default value here instead. So it was postponed and moved out of the scope of this PR - we'll have some room to do that before 1.7 stable without breaking compatibility

File: compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/MultiLayerComposeScene.skiko.kt [hunk 2]
```diff
@@ -16,71 +16,25 @@
...
- * @param composeSceneContext The context to share resources between multiple scenes and provide
- * a way for platform interaction.
- * @param invalidate The function to be called when the content need to be recomposed or
- * re-rendered. If you draw your content using [ComposeScene.render] method, in this callback you
- * should schedule the next [ComposeScene.render] in your rendering loop.
- * @return The created [ComposeScene].
- *
- * @see ComposeScene
- */
+@Deprecated(
```
Comment (igordmn) [inline compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/MultiLayerComposeScene.skiko.kt:26]: I would just remove it, as we I agreed to keep this API public, but as `@InternalComposeUiApi`.

Comment (MatkovIvan) [reply compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/MultiLayerComposeScene.skiko.kt:26]: I agree that we didn't commit to keeping compatibility for such APIs, but I'd remove it after 1.7 to let people migrate easier (IDE replaces usage based on this annotation). We know that this particular one is actually used in some cases

**Review by igordmn — APPROVED** [review]

**Pull request merged**